File: app-presentation/src/main/java/com/zxjdev/smile/presentation/common/base/activity/BaseActivity.java
package com.zxjdev.smile.presentation.common.base.activity;

import android.content.Context;
import android.widget.Toast;

import com.zxjdev.smile.presentation.common.util.image.GlideImageLoader;
import com.zxjdev.smile.presentation.common.util.image.ImageLoader;

import dagger.android.support.DaggerAppCompatActivity;

public abstract class BaseActivity extends DaggerAppCompatActivity {

  protected Context context = this;

  protected void showToast(String message) {
    Toast.makeText(context, message, Toast.LENGTH_SHORT).show();
  }

  private ImageLoader imageLoader;

  protected ImageLoader getImageLoader() {
    if (imageLoader == null) {
      imageLoader = new GlideImageLoader(this);
    }
    return imageLoader;
  }
}

File: app-presentation/src/main/java/com/zxjdev/smile/presentation/common/base/activity/BaseActivity.kt
package com.zxjdev.smile.presentation.common.base.activity

import android.content.Context
import android.widget.Toast

import com.zxjdev.smile.presentation.common.util.image.GlideImageLoader
import com.zxjdev.smile.presentation.common.util.image.ImageLoader

import dagger.android.support.DaggerAppCompatActivity

abstract class BaseActivity : DaggerAppCompatActivity() {

    protected var context: Context = this

    protected val imageLoader: ImageLoader by lazy {
        GlideImageLoader(this)
    }

    protected fun showToast(message: String) {
        Toast.makeText(context, message, Toast.LENGTH_SHORT).show()
    }
}
